davis: init at 4.4.1

Ramblurr opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

[davis](https://github.com/tchapi/davis) is a simple CardDav and CalDav server. It is written in PHP, and uses the Symfony stack. This PR contains 

1. The `davis` package itself
2. and a `services.davis` nixos module

~~**Note** that I am not in the maintainers list yet (waiting for https://github.com/NixOS/nixpkgs/pull/296852 to get merged), so I have temporarily left the pkg maintainers list empty. Once that PR is merged I'll update it.~~ Updated.

Result of `nixpkgs-review pr 298938` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>davis</li>
  </ul>
</details>

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Ramblurr committed:
davis: init at 4.4.1

Release: https://github.com/tchapi/davis/releases/tag/v4.4.1

Ramblurr committed:
nixos/davis: init

Ramblurr commented:
Added a nixos module test, no further changes expected.

Ramblurr commented:
Added myself as maintainer, and added a nixos module test, no further changes expected.

superherointj commented on pkgs/by-name/da/davis/package.nix:
```diff
@@ -0,0 +1,43 @@
+    # The .env files contain runtime configuration that shouldn't be baked into deriviation for the package
+    # This patch adds a few extension points exposing three environment variables:
+    #    RUNTIME_DIRECTORY (where to load .env from), CACHE_DIRECTORY and LOG_DIRECTORY (symfony cache and log rw directories)
+    ./davis-data.patch
```
What is the source of this patch? If the source is in AUR or some other site like that we usually use fetchpatch, goal is to avoid adding patche files  to nixpkgs unless necessary. Then, it's important to provide the reference of source material (as in a comment with url).

Ramblurr replied:
I wrote the patch, so the source is me.

Ramblurr replied:
.. added a link to the upstream PR

superherointj commented on pkgs/by-name/da/davis/package.nix:
```diff
@@ -0,0 +1,43 @@
+  '';
+
+  composerLock = ./composer.lock;
+  vendorHash = "sha256-WGeNwBRzfUXa7kPIwd7/5dPXDjaBxXirAJcm6lNzueY=";
```
```suggestion
  vendorHash = "sha256-WGeNwBRzfUXa7kPIwd7/5dPXDjaBxXirAJcm6lNzueY=";
```
Please move the hash to just after `src`. Notice the hash may change when updating the package. So it is good to have at the top. Also move composerLock to top (it would be fetch phase, right?).

superherointj commented:
It's missing from the body of commit message:

> Release: https://github.com/tchapi/davis/releases/tag/v4.4.1

On upgrade PRs, also add:

> Changelog: https://github.com/tchapi/davis/compare/v4.4.0...v4.4.1

Ramblurr commented:
> It's missing from the body of commit message:
> 
> > Release: https://github.com/tchapi/davis/releases/tag/v4.4.1
> 
> On upgrade PRs, also add:
> 
> > Changelog: [tchapi/davis@v4.4.0...v4.4.1](https://github.com/tchapi/davis/compare/v4.4.0...v4.4.1)
> 
> (reviewers/commiters might read upstream source code changes)

Sorry I don't understand.. what is "it" ?

superherointj commented:
> Sorry I don't understand.. what is "it" ?

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

Link to release note is missing.

superherointj commented:
@Ramblurr Did you understand the missing release pink in commit body message?

Ramblurr commented:
> Did you understand the missing release link in commit body message?

Yes I understand now :) Just pushed that change. Thanks for pointing that out.

superherointj commented on nixos/modules/services/web-apps/davis.nix:
```diff
@@ -0,0 +1,499 @@
+
+  meta = {
+    doc = ./davis.md;
+    maintainers = with lib.maintainers; [ ramblurr ];
```
```suggestion
    maintainers = with lib.maintainers; [ ramblurr ];
```
Consider syncing to package. (`davis.meta.maintainers`)

superherointj commented on nixos/tests/davis.nix:
```diff
@@ -0,0 +1,57 @@
+  {
+    name = "davis";
+
+    meta.maintainers = with lib.maintainers; [ ];
```
```suggestion
    meta.maintainers = with lib.maintainers; [ ];
```
This could be in sync with package that this is testing: `davis.meta.maintainers`
If someone only wants to not follow a certain file, it is possible to filter them out of it. This way, a person that is interested in package, only needs to add themselves once, and follow the tests and module. I think this pattern better. While, the optionality exist.

Ramblurr commented:
I've updated the maintainer lines in the nixos module and test to point to the pkg maintainer (me).

The formatting has changed because I switched to using [nixfmt-rfc-style](https://search.nixos.org/packages?channel=unstable&show=nixfmt-rfc-style&from=0&size=50&sort=relevance&type=packages&query=nixfmt), the upcoming (soon to be?) standard formatter.

Ramblurr commented:
Upstream is accepting our patch, but renamed the env vars. Pushed a change that converges on their naming. Soon we will be able to drop our patch.
